src/Mvc/Mvc.ViewFeatures/src/ViewDataDictionary.cs

```diff
@@ -391,18 +391,30 @@ public ICollection<string> Keys
     public string? Eval(string? expression, string? format)
     {
         var value = Eval(expression);
-        return FormatValue(value, format);
+        return FormatValue(value, format, CultureInfo.CurrentCulture);
     }
 
     /// <summary>
-    /// Formats the given <paramref name="value"/> using given <paramref name="format"/>.
+    /// Formats the given <paramref name="value"/> using the given <paramref name="format"/>.
     /// </summary>
     /// <param name="value">The value to format.</param>
     /// <param name="format">
     /// The format string (see <see href="https://msdn.microsoft.com/en-us/library/txafckwd.aspx"/>).
     /// </param>
     /// <returns>The formatted <see cref="string"/>.</returns>
     public static string? FormatValue(object? value, string? format)
+        => FormatValue(value, format, CultureInfo.CurrentCulture);
+
+    /// <summary>
+    /// Formats the given <paramref name="value"/> using the given <paramref name="format"/> and <paramref name="formatProvider"/>.
+    /// </summary>
+    /// <param name="value">The value to format.</param>
+    /// <param name="format">
+    /// The format string (see <see href="https://msdn.microsoft.com/en-us/library/txafckwd.aspx"/>).
+    /// </param>
+    /// <param name="formatProvider">The <see cref="IFormatProvider"/> used to format the value.</param>
+    /// <returns>The formatted <see cref="string"/>.</returns>
+    public static string? FormatValue(object? value, string? format, IFormatProvider formatProvider)
```

Comment:
This new method (as well as some of the other public API additions) could possibly be made `internal`. We need to decide which of these new APIs (if any) should be public.